src/Microsoft.AspNet.Mvc.Core/Filters/FormatFilterAttribute.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.AspNet.Mvc.Description;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.OptionsModel;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// A filter which will use the format value in the route data or query string to set the content type on an 
+    /// <see cref="ObjectResult" /> returned from an action.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
+    public class FormatFilterAttribute : Attribute, IFormatFilter, IResourceFilter, IResultFilter
+    {
+        /// <summary>
+        /// As a <see cref="IResourceFilter"/>, this filter looks at the request and rejects it before going ahead if
+        /// 1. The format in the request doesnt match any format in the map.
+        /// 2. If there is a conflicting producesFilter.
+        /// </summary>
+        /// <param name="context">The <see cref="ResourceExecutingContext"/>.</param>
+        public void OnResourceExecuting([NotNull] ResourceExecutingContext context)
+        {
+            var format = GetFormat(context);
+
+            if (!string.IsNullOrEmpty(format))
+            {
+                var formatContentType = GetContentType(format, context);
+                if (formatContentType == null)
+                {
+                    // no contentType exists for the format, return 404
+                    context.Result = new HttpNotFoundResult();
+                }
+                else
+                {
+                    var responseTypeFilters = context.Filters.OfType<IApiResponseMetadataProvider>();                    
+                    var contentTypes = new List<MediaTypeHeaderValue>();
+
+                    foreach (var filter in responseTypeFilters)
+                    {
+                        filter.SetContentTypes(contentTypes);
+                    }
+
+                    if (contentTypes.Count != 0)
+                    {
+                        // If formatfilterContentType is not subset of any of the content types produced by 
+                        // IApiResponseMetadataProviders, return 404
+                        if (!contentTypes.Any(c => formatContentType.IsSubsetOf(c)))
+                        {
+                            context.Result = new HttpNotFoundResult();
+                        }
+                    }                   
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        public void OnResourceExecuted([NotNull] ResourceExecutedContext context)
+        {
```

Comment:
Remove the unnecessary spacing, follow the style in `Controller.OnActionExecuting`